### DISCUSSION TITLE
Are any of these features available in Helix?

### DISCUSSION THREAD

**Discussion by dcompoze** [helix-editor/helix#3156] [Q&A] [answered]
Could someone help me figure out if any of these features are already present in Helix or if there are alternative ways of achieving the same functionality before I delve deeper into the editor's codebase?

- Completion of words from current buffer 
    - **Answered:** #2403

<img src="https://user-images.githubusercontent.com/76269710/180611758-c7f44cb2-9571-4b6b-9061-c185802b5ab4.png" width="25%">

- Completion of local file paths in the current buffer 
    - **Answered:** #2608

<img src="https://user-images.githubusercontent.com/76269710/180611776-4f7290fe-87d8-48a1-97a1-f293acb75fb1.png" width="60%">

- A way to see the location relative to the total size of the file (e.g. scroll bar or percentage)

<img src="https://user-images.githubusercontent.com/76269710/180611793-eb002050-3296-4422-bd3f-5c7df8606117.png" width="45%">

- A way to enter unicode codepoints directly (e.g. vim's `Ctrl-v` in insert mode)

https://user-images.githubusercontent.com/76269710/180611806-7dad1aba-27ff-490e-a33a-70012e0649c0.mp4

- Select visual block (I'm aware of multiple cursor mode, but it seems not to select a strict rectangle)

Vim block selection:

https://user-images.githubusercontent.com/76269710/180611210-8d80e298-588c-4c51-a836-788424ad9458.mp4

Helix block selection:

https://user-images.githubusercontent.com/76269710/180611221-3aa4a212-4634-40e5-a491-ab72d31f3c5c.mp4

- Undo cache (e.g. in `vim` you can undo changes after closing a file) 
    - **Answered:** #401

- Dynamically reload the document in the buffer when it changes on disk 
    - **Answered:** #1125

- Snippet support (e.g. Ultisnips) 
    - **Answered:** #395

<img src="https://user-images.githubusercontent.com/76269710/180612283-d4a0bd6e-a03d-42f6-a7d7-a38696bfaaa6.png" width="65%">

- Spell checker integration (e.g. `aspell`)

<img src="https://user-images.githubusercontent.com/76269710/180612276-3c509c47-7bf8-49ba-959f-7eab524ffcfc.png" width="40%">

- Git mergetool integration (e.g. I'm currently using `neovim` + `vim-fugitive` as a git `mergetool`) 
    - **Answered:** #227

- Git added/modified/removed indicators in the gutter 
    - **Answered:** #1623

- Highlighting color values (e.g. `#FF00FF` or `rgb(255, 0, 255)`) with a corresponding background color or indicator.

<img src="https://user-images.githubusercontent.com/76269710/180612268-a7666c46-12cf-4204-a9f1-cb41475e1f7b.png" width="40%">

**Comment by kirawi** [comment, marked as answer]
> Completion of words from current buffer

#2403

> Completion of local file paths in the current buffer

#2608

> Undo cache (e.g. in vim you can undo changes after closing a file)

#401 

> Dynamically reload the document in the buffer when it changes on disk

#1125 

> Snippet support (e.g. Ultisnips)

#395

> Git mergetool integration (e.g. I'm currently using neovim + vim-fugitive as a git mergetool)

#227 

> Git added/modified/removed indicators in the gutter

#1623

**Comment by sbromberger** [comment]
> A way to see the location relative to the total size of the file (e.g. scroll bar or percentage)

#3168

**Comment by Rudxain** [comment]
>     * Highlighting color values (e.g. `#FF00FF` or `rgb(255, 0, 255)`) with a corresponding background color or indicator.

- #1115
- #12308
